test/blas.jl

```diff
@@ -51,252 +50,260 @@ for elty in [Float32, Float64, Complex64, Complex128]
     v41 = convert(Vector{elty}, [4:-1:1;])
 
     let n = 10
-        # dot
-        if elty <: Real
-            x1 = convert(Vector{elty}, randn(n))
-            x2 = convert(Vector{elty}, randn(n))
-            @test BLAS.dot(x1,x2) ≈ sum(x1.*x2)
-            @test_throws DimensionMismatch BLAS.dot(x1,rand(elty, n + 1))
-        else
-            z1 = convert(Vector{elty}, complex.(randn(n),randn(n)))
-            z2 = convert(Vector{elty}, complex.(randn(n),randn(n)))
-            @test BLAS.dotc(z1,z2) ≈ sum(conj(z1).*z2)
-            @test BLAS.dotu(z1,z2) ≈ sum(z1.*z2)
-            @test_throws DimensionMismatch BLAS.dotc(z1,rand(elty, n + 1))
-            @test_throws DimensionMismatch BLAS.dotu(z1,rand(elty, n + 1))
+        @testset "dot products" begin
+            if elty <: Real
+                x1 = convert(Vector{elty}, randn(n))
+                x2 = convert(Vector{elty}, randn(n))
+                @test BLAS.dot(x1,x2) ≈ sum(x1.*x2)
+                @test_throws DimensionMismatch BLAS.dot(x1,rand(elty, n + 1))
+            else
+                z1 = convert(Vector{elty}, complex.(randn(n),randn(n)))
+                z2 = convert(Vector{elty}, complex.(randn(n),randn(n)))
+                @test BLAS.dotc(z1,z2) ≈ sum(conj(z1).*z2)
+                @test BLAS.dotu(z1,z2) ≈ sum(z1.*z2)
+                @test_throws DimensionMismatch BLAS.dotc(z1,rand(elty, n + 1))
+                @test_throws DimensionMismatch BLAS.dotu(z1,rand(elty, n + 1))
+            end
         end
-
-        #iamax
-        if elty <: Real
-            x = convert(Vector{elty}, randn(n))
-            @test BLAS.iamax(x) == indmax(abs.(x))
-        else
-            z = convert(Vector{elty}, complex.(randn(n),randn(n)))
-            @test BLAS.iamax(z) == indmax(map(x -> abs(real(x)) + abs(imag(x)), z))
+        @testset "iamax" begin
+            if elty <: Real
+                x = convert(Vector{elty}, randn(n))
+                @test BLAS.iamax(x) == indmax(abs.(x))
+            else
+                z = convert(Vector{elty}, complex.(randn(n),randn(n)))
+                @test BLAS.iamax(z) == indmax(map(x -> abs(real(x)) + abs(imag(x)), z))
+            end
         end
-
-        # axpy
-        if elty <: Real
-            x1 = convert(Vector{elty}, randn(n))
-            x2 = convert(Vector{elty}, randn(n))
+        @testset "axpy" begin
+            if elty <: Real
+                x1 = convert(Vector{elty}, randn(n))
+                x2 = convert(Vector{elty}, randn(n))
                 α  = rand(elty)
```

Comment:
this alpha could've been reused for both Real and non-Real elty, not that it makes too much difference